signingscript/sign.py

```diff
@@ -228,18 +234,95 @@ def build_signtool_cmd(context, from_, fmt, to=None):
             await sign_file(context, from_, fmt)
     if file_extension == '.zip':
         # Recreate the zipfile
-        await _create_zipfile(context, orig_path, files)
+        await _create_zipfile(context, orig_path, files, tmp_dir=tmp_dir)
     return orig_path
 
 
 # sign_widevine {{{1
 async def sign_widevine(context, orig_path, fmt):
-    """Sign the internals of a file with the widevine key.
+    """Call the appropriate helper function to do widevine signing.
 
-    Extract the zip or tarball and only sign a handful of files (see
+    Args:
+        context (SigningContext): the signing context
+        orig_path (str): the source file to sign
+        fmt (str): the format to sign with
+
+    Raises:
+        SigningScriptError: on unknown suffix.
+
+    Returns:
+        str: the path to the signed archive
+
+    """
+    ext_to_fn = {
```

Comment:
I think I then get an error about missing functions, since the module scope happens above the function definitions. I solved for this earlier by moving functions to the sign module, but then we'd have to have a second helper module.